packages/common/src/store/gated-content/sagas.ts

```diff
@@ -488,14 +553,25 @@ function* updateSpecialAccessTracks(
 
   Object.keys(cachedTracks).forEach((trackId) => {
     const id = parseInt(trackId)
-    const { owner_id: ownerId, stream_conditions: streamConditions } =
-      cachedTracks[id]
-    const isGated =
+    const {
+      owner_id: ownerId,
+      stream_conditions: streamConditions,
+      download_conditions: downloadConditions
+    } = cachedTracks[id]
+    const isTrackStreamGated =
       gate === 'follow'
         ? isContentFollowGated(streamConditions)
         : isContentTipGated(streamConditions)
-    if (isGated && ownerId === trackOwnerId) {
+    const isTrackDownloadGated =
+      gate === 'follow'
+        ? isContentFollowGated(downloadConditions)
+        : isContentTipGated(downloadConditions)
+    if (isTrackStreamGated && ownerId === trackOwnerId) {
       statusMap[id] = 'UNLOCKING'
+      // todo: if necessary, update some ui status to show that the track download is unlocking
```

Comment:
are all these todos necessary? seeing a lot of them. also nit all-caps them so they get highlighted in vscode.